src/tools/compiletest/src/header.rs

```diff
@@ -637,7 +637,11 @@ impl TestProps {
     pub fn pass_mode(&self, config: &Config) -> Option<PassMode> {
         if !self.ignore_pass {
             if let (mode @ Some(_), Some(_)) = (config.force_pass_mode, self.pass_mode) {
-                return mode;
+                if self.pass_mode != PassMode::RunFail {
```

Comment:
`self.pass_mode` is of type `Option<PassMode>` so I doubt this will work.

Reply:
Fixed, I hope